logstreams/src/main/java/io/zeebe/logstreams/impl/log/LogStreamReaderImpl.java

```diff
@@ -7,260 +7,197 @@
  */
 package io.zeebe.logstreams.impl.log;
 
-import io.zeebe.logstreams.impl.Loggers;
 import io.zeebe.logstreams.log.LogStreamReader;
 import io.zeebe.logstreams.log.LoggedEvent;
-import io.zeebe.logstreams.spi.LogStorage;
-import io.zeebe.logstreams.spi.LogStorageReader;
+import io.zeebe.logstreams.storage.LogStorageReader;
 import java.util.NoSuchElementException;
-import java.util.function.LongUnaryOperator;
 import org.agrona.DirectBuffer;
 import org.agrona.concurrent.UnsafeBuffer;
 
+/**
+ * A {@link LogStreamReader} implementation which can read single events from the blocks given by
+ * the {@link LogStorageReader}.
+ *
+ * <p>This implementation assumes that blocks have no padding - they contain a contiguous series of
+ * {@link LoggedEvent} which fits exactly within the block.
+ *
+ * <p>The implementation currently does not perform any copying, however it is a good candidate to
+ * introduce further look-ahead to improve I/O performance which would necessarily require copying.
+ */
```

Comment:
If we do more look ahead, there's a chance that we might span more than one segment, and when we switch segment we will unmap the file, resulting in a segfault, I think. But I guess since we do a single look ahead this is already a chance, so...I can remove this